packages/flutter/lib/src/cupertino/text_form_field_row.dart

```diff
@@ -272,19 +277,45 @@ class CupertinoTextFormFieldRow extends FormField<String> {
 }
 
 class _CupertinoTextFormFieldRowState extends FormFieldState<String> {
-  TextEditingController? _controller;
+  RestorableTextEditingController? _controller;
 
-  TextEditingController? get _effectiveController =>
-      _cupertinoTextFormFieldRow.controller ?? _controller;
+  TextEditingController get _effectiveController =>
+      _cupertinoTextFormFieldRow.controller ?? _controller!.value;
 
   CupertinoTextFormFieldRow get _cupertinoTextFormFieldRow =>
       super.widget as CupertinoTextFormFieldRow;
 
+  @override
+  void restoreState(RestorationBucket? oldBucket, bool initialRestore) {
+    super.restoreState(oldBucket, initialRestore);
+    if (_controller != null) {
+      _registerController();
+    }
+    // Make sure to update the internal [FormFieldState] value to sync up with
+    // text editing controller value.
```

Comment:
Nit:

```
// This makes sure to...
```

I think this is more clear that the comment is just an explanation. When I first read it, I thought it might be a command to callers to make sure that they also update the value.